src/main/java/com/iexec/sms/secret/web2/Web2SecretsService.java

```diff
@@ -50,43 +50,54 @@ public Optional<Secret> getSecret(String ownerAddress, String secretAddress) {
 
     public Optional<Secret> getSecret(String ownerAddress, String secretAddress, boolean shouldDecryptValue) {
         Optional<Secret> oSecret = getSecret(ownerAddress, secretAddress);
-        return shouldDecryptValue ? oSecret.map(this::decryptSecret) : oSecret;
+
+        if (oSecret.isEmpty() || !shouldDecryptValue) {
+            return oSecret;
+        }
+
+        final Secret secret = oSecret.get();
+        final String decryptedValue = encryptionService.decrypt(secret.getValue());
+        return Optional.of(secret.withDecryptedValue(decryptedValue));
     }
 
-    public boolean addSecret(String ownerAddress, String secretAddress, String secretValue) {
+    public Optional<Secret> addSecret(String ownerAddress, String secretAddress, String secretValue) {
```

Comment:
IMHO, it helps to check the method worked correctly. I had a case where I had an issue with the result that wouldn't have been detected by the tests without this `Optional<Secret>` return. I don't say this was introduced for this test, but this is helpful.
The question now is: do we want to support things in production that help testing/debugging? I'd say it can hurt in this case, as we return an immutable object.